src/sentry/digests/backends/dummy.py

```diff
@@ -1,22 +1,36 @@
 from contextlib import contextmanager
+from typing import TYPE_CHECKING, Any, Iterable, Optional
 
 from sentry.digests.backends.base import Backend
 
+if TYPE_CHECKING:
```

Comment:
Why do we need to do this?

Reply:
Because of the `Record` class, there is a circular dependency:
- `/digests/__init__.py`
- `/digests/backends/dummy.py` 
- `/digests/__init__.py` 
We only need `Record` and `ScheduleEntry` to annotate the parameters and return types so we use `TYPE_CHECKING` to break the dependency at runtime.

Reply:
This seems a bad smell to me. If this can be solve by moving some of those types/classes to another file I think that would be better.

Reply:
I'm trying not to move functions and files around in the typing PR.